include/oneapi/dpl/internal/random_impl/linear_congruential_engine.h

```diff
@@ -230,8 +230,7 @@ class linear_congruential_engine
     {
         ::std::uint64_t __mod = modulus;
         ::std::uint64_t __mult = pow_mult_n(__num_to_skip);
-        for (int __i = 0; __i < _N; ++__i)
-            state_[__i] = static_cast<scalar_type>(((__mult * static_cast<::std::uint64_t>(state_[__i])) % __mod));
+        state_ = ((__mult * state_.template convert<::std::uint64_t>()) % __mod).template convert<scalar_type>();
```

Comment:
How does it affect performance on different devices?

Reply:
I'm collecting performance data right now, will update the comment after I finish